test/Microsoft.OpenApi.Tests/Validations/OpenApiReferenceValidationTests.cs

```diff
@@ -0,0 +1,164 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license. 
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.OpenApi.Extensions;
+using Microsoft.OpenApi.Interfaces;
+using Microsoft.OpenApi.Models;
+using Microsoft.OpenApi.Validations;
+using Xunit;
+
+namespace Microsoft.OpenApi.Tests.Validations
+{
+    public class OpenApiReferenceValidationTests
+    {
+        [Fact]
+        public void ReferencedSchemaShouldOnlyBeValidatedOnce()
+        {
+            // Arrange
+
+            var sharedSchema = new OpenApiSchema
+            {
+                Type = "string",
+                Reference = new OpenApiReference()
+                {
+                    Id = "test"
+                },
+                UnresolvedReference = false
+            };
+
+            OpenApiDocument document = new OpenApiDocument();
+            document.Components = new OpenApiComponents()
+            {
+                Schemas = new Dictionary<string, OpenApiSchema>()
+                {
+                    [sharedSchema.Reference.Id] = sharedSchema
+                }
+            };
+
+            document.Paths = new OpenApiPaths()
+            {
+                ["/"] = new OpenApiPathItem()
+                {
+                    Operations = new Dictionary<OperationType,OpenApiOperation>
+                    {
+                        [OperationType.Get] = new OpenApiOperation()
+                        {
+                            Responses = new OpenApiResponses()
+                            {
+                                ["200"] = new OpenApiResponse()
+                                {
+                                    Content = new Dictionary<string,OpenApiMediaType>()
+                                    {
+                                        ["application/json"] = new OpenApiMediaType()
+                                        {
+                                            Schema = sharedSchema
+                                        }
+                                    }
+                                }
+                            }
+                        }
+                    }
+                }
+            };
+
+            // Act
+            var errors = document.Validate(new ValidationRuleSet() { new AlwaysFailRule<OpenApiSchema>()});
+
+
+            // Assert
+            Assert.True(errors.Count() == 1);            
+        }
+
+        [Fact]
+        public void UnResolvedReferencedSchemaShouldNotBeValidated()
```

Comment:
>UnResolvedReferencedSchemaShouldNotBeValidated [](start = 20, length = 46)

nit, casing of unresolved incorrect:

UnresolvedReferencedSchemaShouldNotBeValidated

Reply:
Fixed the casing issue.  One tests is ReferencedSchema, the other is SchemaReference

---
In reply to: [184120408](https://github.com/Microsoft/OpenAPI.NET/pull/241#discussion_r184120408) [](ancestors = 184120408)

Reply:
Ah, ok. That makes sense. The other one has a typo though ("Referenced" instead of "Reference")

---
In reply to: [184126122](https://github.com/Microsoft/OpenAPI.NET/pull/241#discussion_r184126122) [](ancestors = 184126122,184120408)

Reply:
"Referenced" was intentional, but Reference works too.  Changed it.

---
In reply to: [184126692](https://github.com/Microsoft/OpenAPI.NET/pull/241#discussion_r184126692) [](ancestors = 184126692,184126122,184120408)